### PR TITLE
Feature: Db size in bytes

### PR DESCRIPTION
**Pull request by kwek20**
See #1653

### CODE REVIEW

File: src/main/java/com/iota/iri/Iota.java [hunk 1]
```diff
@@ -22,22 +8,34 @@
 import com.iota.iri.network.TransactionRequester;
 import com.iota.iri.network.pipeline.TransactionProcessingPipeline;
 import com.iota.iri.service.ledger.LedgerService;
-import com.iota.iri.service.milestone.LatestMilestoneTracker;
-import com.iota.iri.service.milestone.LatestSolidMilestoneTracker;
-import com.iota.iri.service.milestone.MilestoneService;
-import com.iota.iri.service.milestone.MilestoneSolidifier;
-import com.iota.iri.service.milestone.SeenMilestonesRetriever;
+import com.iota.iri.service.milestone.*;
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java [hunk 2]
```diff
@@ -548,7 +569,7 @@ private void initClassTreeMap(List<ColumnFamilyDescriptor> columnFamilyDescripto
         classTreeMap = MapUtils.unmodifiableMap(classMap);
     }
 
-    private DBOptions createOptions(String logPath, String configFile) throws IOException {
+    private DBOptions createOptions(String logPath, String configFile, List<ColumnFamilyDescriptor> columnFamilyDescriptors) throws IOException {
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java [hunk 3]
```diff
@@ -101,4 +109,61 @@ public static ExecutorService createNamedSingleThreadExecutor(String name) {
...
+	
+	/**
+     * Parses a human readable string for a file size (tb, gb, mb -> file modifier)
+     * Follows the following format: [double][optional space][case insensitive file modifier]
+     * 
+     * Kb is not parsed as this is too small for us to be used in a persistence provider
+     * 
+     * @return The size of the human readable string in bytes, rounded down.
+     */
+    public static long parseFileSize(String humanReadableSize) {
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/service/snapshot/impl/LocalSnapshotManagerImpl.java [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 
-import com.google.common.annotations.VisibleForTesting;
+import com.iota.iri.conf.BaseIotaConfig;
 import com.iota.iri.conf.SnapshotConfig;
 import com.iota.iri.service.milestone.LatestMilestoneTracker;
-import com.iota.iri.service.snapshot.LocalSnapshotManager;
-import com.iota.iri.service.snapshot.SnapshotService;
-import com.iota.iri.service.snapshot.SnapshotException;
-import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.*;
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/snapshot/impl/LocalSnapshotManagerImpl.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.snapshot'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

**Comment by GalRogozinski** [comment]
Still waiting:
[ ] - For answering all manual comments on #1653 
[ ] - Unblocking buildkite
[ ] - Changing the magic number 5

**Comment by kwek20** [comment]
^done

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/utils/IotaUtils.java]: Doesn't look too bad

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java [hunk 5]
```diff
@@ -607,4 +644,9 @@ private DBOptions createOptions(String logPath, String configFile) throws IOExce
 
         return options;
     }
+    
+    @Override
+    public long getPersistanceSize() {
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: This is a little thing that annoys me (found out just now)...
You have a Typo  -> Persistence

Comment (kwek20) [reply src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: uhhhhg every time i type that word wrong :p

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/Iota.java (hunk 1 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/Iota.java]: Codacy Issue found: [Avoid unused imports such as 'com.iota.iri.service.milestone'](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/utils/IotaUtils.java (hunk 3 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/utils/IotaUtils.java]: Codacy Issue found: [The method 'parseFileSize(String)' has a cyclomatic complexity of 13.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has an NPath complexity of 504](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: Codacy Issue found: [The method 'createOptions(String, String, List)' has a cyclomatic complexity of 16.](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4857464)

File: src/main/java/com/iota/iri/service/transactionpruning/SizePruningCondition.java [hunk 6]
```diff
@@ -88,7 +88,7 @@ public int getSnapshotPruningMilestone() throws TransactionPruningException {
         int initialIndex;
         try {
             Pair<Indexable, Persistable> ms = tangle.getFirst(Milestone.class, IntegerIndex.class);
-            if (ms == null) {
+            if (ms == null || ms.low == null) {
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/transactionpruning/SizePruningCondition.java]: A question, the current annoying code actually returns null?
This is surprising I thought it just returned an empty object...

Comment (kwek20) [reply src/main/java/com/iota/iri/service/transactionpruning/SizePruningCondition.java]: no it doesnt

Comment (kwek20) [reply src/main/java/com/iota/iri/service/transactionpruning/SizePruningCondition.java]: to clarify, ms is never null, ms.low can be null

**Pull request merged**